zap/src/main/java/org/parosproxy/paros/core/scanner/Variant.java

```diff
@@ -47,6 +48,14 @@ public String setParameter(
     public String setEscapedParameter(
             HttpMessage msg, NameValuePair originalPair, String param, String value);
 
+    /**
+     * Sets the parameters into the given {@code message}.
+     *
+     * @param message the message that will be changed
+     * @param appParams list of name of the parameter
+     */
+    default void setParameters(HttpMessage message, List<InputVector> appParams) {};
```

Comment:
- Add since tag, same for the new methods in the implementation.
- `appParams` needs to be renamed.

Reply:
but every method is set*.

Reply:
will add since tag

Reply:
Sorry, is that referring to which comment?

Reply:
Ok, I see the edit, as mentioned earlier they might not be "app" parameters.

Reply:
oh sorry for the confusion.